keps/sig-auth/20190607-certificates-api.md

```diff
@@ -234,7 +261,31 @@ Cluster admins can either:
 1. grant signer-specific approval permissions using roles they define
 2. grant signer-specific approval permissions using the bootstrap roles starting in 1.18
 3. disable the approval-authorizing admission plugin in 1.18 (if they don't care about partitioning approver rights)
- 
+
+### API changes between v1beta1 and v1
+
+`spec.signerName` validation:
+
+* required, not defaulted
+* `kubernetes.io/legacy-unknown` is disallowed when creating new CSR objects
+
+`status.certificate` validation:
+
+* cannot be unset or changed once set
+* must contain one or more PEM blocks
+* all PEM blocks must have the "CERTIFICATE" label, contain no headers, and the encoded data
+  must be a BER-encoded ASN.1 Certificate structure as described in section 4 of RFC5280.
+* non-PEM content may appear before or after the CERTIFICATE PEM blocks and is unvalidated,
+  to allow for explanatory text as described in section 5.2 of RFC7468.
+
+`status.conditions` validation:
+
+* type is required
+* status is required and must be `True`, `False`, or `Unknown`
+
+`/status` subresource:
+
+* allows modifying `status.conditions` other than `Approved`/`Denied`
```

Comment:
why wouldn't we use the general status subresource for setting approved and denied conditions too?  Someone already had to opt-in for v1 and they already have to understand /status to set the signed certificate.  it doesn't seem like too much to ask. In fact, it makes it atomic.

Reply:
authz policy isn't version-aware. `certificatesigningrequests/status` would mean different things for v1 and v1beta1.

Reply:
> authz policy isn't version-aware. `certificatesigningrequests/status` would mean different things for v1 and v1beta1.

So it would.  Explain why it isn't an option so the future reader like me knows why it looks weird, but needs to stay?

Reply:
sure